canvas/image.go

```diff
@@ -74,6 +79,49 @@ func NewImageFromFile(file string) *Image {
 	}
 }
 
+// NewImageFromURI creates a new image from named resource.
+// File URIs will read the file path and other schemes will download the data into a resource
+// Images returned from this method will scale to fit the canvas object.
+// The method for scaling can be set using the Fill field.
+func NewImageFromURI(uri fyne.URI) *Image {
+	if uri.Scheme() == "file" && len(uri.String()) > 7 {
+		return &Image{
+			File: uri.String()[7:],
+		}
+	}
+
+	var read io.ReadCloser
```

Comment:
Could this be split out into `NewImageFromReader`?

Reply:
A cunning plan. Does the rest of the PR seem sensible? I was more keen to get feedback on the http/https part in the  description.

Reply:
Added